Gems/Multiplayer/Code/Include/Multiplayer/Components/SimplePlayerSpawnerComponentBus.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/std/string/string.h>
+
+namespace Multiplayer
+{
+    //! @class SimplePlayerSpawnerRequests
+    //! @brief The SimplePlayerSpawnerRequest event-bus exposes helper methods regarding network player spawners.
+    //! Although the Multiplayer System automatically spawns in players, it's common for game specific server logic to retrieve valid spawn locations when respawning a player.
+    class SimplePlayerSpawnerRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        
+        //! Returns the location where the next joining player will be spawned, and then advances the spawn point index.
+        //! The index will wrap back to 0 in a round-robin fashion so that a valid spawn point is always available.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding the spawn point index.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform RoundRobinNextSpawnPoint() = 0;
+
+        //! Returns the location where the next joining player will be spawned. Unlike RoundRobinNextSpawnPoint(), this will not cause the current spawn point index to increment.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding the spawn point index.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform GetNextSpawnPoint() const = 0;
+
+
+        //! Returns a mutable list of all the spawn points.
+        //! Only access/edit this list on the multiplayer host; spawn points are not synced across the network and the host is responsible for spawning players.
+        //! @return AZStd::vector<AZ::EntityId> List of spawn points.
+        virtual AZStd::vector<AZ::EntityId>& GetSpawnPoints() = 0;
```

Comment:
Suggestion: This should probably be an _immutable_ list. It's a bit dangerous to expose the internal storage of the component directly outwards through the API. For example, suppose the component needed to keep a second data structure in sync with this list. Like maybe a "num times used" counter per spawn point, to support a "semi-random" selection that ensures that every point is used once in a rotation before using it the next time. If the list itself is exposed outwards, there's no way to build a second data structure like that and keep it in sync.

Reply:
Good call! Fixed!